### PR TITLE
Separate UI from data storage

### PR DESCRIPTION
**Pull request by mduczmal**
2 test failing: testLoadAndUpdateItem, testInsertAndUpdateItem

### CODE REVIEW

File: src/main/java/io/domain/Store.java [hunk 1]
```diff
@@ -1,5 +1,6 @@
 package io.domain;
 
+
```
Comment (walczakb) [inline src/main/java/io/domain/Store.java]: Please remove the empty line.

**Review by walczakb — CHANGES_REQUESTED** [review]

File: src/main/java/io/domain/Store.java (hunk 1 above)
Comment (maciejnems) [inline src/main/java/io/domain/Store.java]: Needless empty line

File: src/main/java/io/domain/Store.java [hunk 2]
```diff
@@ -18,7 +19,12 @@
 
     public List<Item> items() { return Collections.unmodifiableList(items); }
 
-    public void addItem(Item item) {
+    protected Item createItem(String name, int count, int price) {
+        return new Item(name, count, price);
+    }
+
+    public void addItem(String name, int count, int price) {
```
Comment (maciejnems) [inline src/main/java/io/domain/Store.java]: Method addItem should return created Item object

File: src/main/java/io/swingui/WindowFactory.java [hunk 3]
```diff
@@ -1,29 +1,27 @@
...
 import io.domain.Store;
 import io.ui.ItemPresenter;
 import io.ui.StorePresenter;
 import io.ui.ViewFactory;
 
 public class WindowFactory implements ViewFactory {
-    private final DbGateway db;
 
-    public WindowFactory(DbGateway db) { this.db = db; }
+    public WindowFactory() {}
```
Comment (maciejnems) [inline src/main/java/io/swingui/WindowFactory.java]: Empty constructor can be removed

File: src/main/java/io/db/Database.java [hunk 4]
```diff
@@ -1,35 +1,17 @@
...
 
     public Store loadStore() {
-        Store store = new Store();
-        ids = new HashMap<>();
+        Store store = new StoreProxy(engine);
         engine.readItems((id, name, count, price) -> {
-            Item item = new Item(name, count, price);
-            store.addItem(item);
-            ids.put(item, id);
+            store.addItem(name, count, price);
```
Comment (maciejnems) [inline src/main/java/io/db/Database.java]: This doubles your items that are already in database

**Pull request closed without merging**